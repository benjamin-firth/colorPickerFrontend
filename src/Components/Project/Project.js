import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { apiRequest } from '../../utils/api';
import { deleteProject } from '../../actions/index';
import './Project.scss';
import Palette from '../Palette/Palette';
import PropTypes from 'prop-types';
import { setAllPalettes } from '../../actions'

function Project({ projectInfo, palettes }) {
  const dispatch = useDispatch();
  const { name, id } = projectInfo;

  const displayPalettes = () => {
    const projectPalettes = palettes.filter(palette => parseInt(palette.project_id) === id);

    if (!projectPalettes.length) {
      return <p>No palettes</p>;
    }
    return projectPalettes.map(palette => <Palette palette={palette}/>)
  };

  const handleClick = async () => {
    const body = projectInfo;
    try {
      const response = await apiRequest(`projects`, 'DELETE', body);
      console.log(response);
    } catch (error) {
      console.log(error);
    }
    dispatch(deleteProject(body.id));
  }

  return (
    <section>
      <h3>{name}</h3>
      <button 
        className='delete-palette-button'
        onClick={handleClick}>X
      </button>
      {displayPalettes()}
    </section>
  );
};

export default Project;
