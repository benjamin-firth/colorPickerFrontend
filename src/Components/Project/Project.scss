.saved-project-container {
  margin-bottom: 4vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  p {
    margin: 0;
    margin-top: .4vw;
    font-size: 1.8vw;
  }
  -webkit-column-break-inside: avoid;
  page-break-inside: avoid;
  break-inside: avoid-column;
}

.saved-project-title {
  display: flex;
  justify-content: space-around;
  align-items: center;
  border: 2px solid black;
  width: 17vw;
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
  box-shadow: 5px 5px;
  button {
    border: none;
    font-size: 1.8vw;
  }
  h3 {
    font-size: 1.8vw;
    font-weight: bold;
    margin: 0;
  }
}
