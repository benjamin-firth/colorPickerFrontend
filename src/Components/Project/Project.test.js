import React from 'react';
import { render } from '@testing-library/react';
import Project from './Project';
import { shallow } from 'enzyme';

jest.mock("react-redux", () => ({
  useSelector: () => [{name: "Test", id: 1}],
  useDispatch: () => jest.fn()
}));

let mockProjectInfo = {
  name: 'Test',
  id: 12
}

let mockPalettes = [{
  name: 'test',
  project_id: 2,
  color1: 'FFF',
  color2: 'FFF',
  color3: 'FFF',
  color4: 'FFF',
  color5: 'FFF',
  id: 3
}, {
  name: 'test',
  project_id: 2,
  color1: 'FFF',
  color2: 'FFF',
  color3: 'FFF',
  color4: 'FFF',
  color5: 'FFF',
  id: 3
}];

describe('Project', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Project projectInfo={mockProjectInfo} palettes={mockPalettes}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
