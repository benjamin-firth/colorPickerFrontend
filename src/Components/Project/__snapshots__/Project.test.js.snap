// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot 1`] = `
<section
  className="saved-project-container"
>
  <div
    className="saved-project-title"
  >
    <h3>
      Test
    </h3>
    <button
      className="delete-palette-button"
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <p>
    No palettes
  </p>
</section>
`;
